lib/situation_room/site.ex

```diff
@@ -7,10 +7,31 @@ defmodule SituationRoom.Site do
     field :name, :string
   end
 
+  # Default changeset to be used to validate against schema
   def changeset(site, params) do
     site
     |> cast(params, [:endpoint, :name])
     |> validate_required([:endpoint, :name])
+    |> validate_url(:endpoint)
+  end
+
+  # Function to test if a url is valid and returns why it is not valid
+  def validate_url(changeset, field, opts \\ []) do
+    validate_change changeset, field, fn _, value ->
+      case URI.parse(value) do
+        %URI{scheme: nil} -> "is missing a scheme (e.g. https)"
+        %URI{host: nil} -> "is missing a host"
+        %URI{host: host} ->
+          case :inet.gethostbyname(Kernel.to_charlist host) do
+            {:ok, _} -> nil
+            {:error, _} -> "invalid host"
+          end
+      end
+      |> case do
+        error when is_binary(error) -> [{field, Keyword.get(opts, :message, error)}]
+        _ -> []
+      end
+    end
```

Comment:
Overall, I like this as a first pass, but I think we can probably improve on this a bit; both functionally and also idiomatically from an Elixir perspective.

First, we kind of doing two things here: 1) validating the format of the URI passed in (via using `URI.parse`) and 2) validating that the URI provided actually exists and is reachable (via `:inet.gethostbyname`)

I'm wondering if we should split these into two different validations?  Not sure on that yet, but looking for your thoughts.

For idiomatic elixir, we seem to be directly piping one case statement into another and it's a bit hard to follow.  I think we can better leverage a few languages features like pattern matching, or with clauses and things.  We can sit together and work through some of that if you'd prefer.